### PR TITLE
Remove `netty-all` dependency

### PR DESCRIPTION
**Pull request by EarthChen**
## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-dependencies-bom/pom.xml [hunk 1]
```diff
@@ -203,11 +212,57 @@
...
+<!--                <scope>runtime</scope>-->
+<!--            </dependency>-->
+<!--            <dependency>-->
+<!--                <groupId>io.netty</groupId>-->
+<!--                <artifactId>netty-transport-native-kqueue</artifactId>-->
+<!--                <version>${netty4_version}</version>-->
+<!--                <classifier>osx-aarch_64</classifier>-->
+<!--                <scope>runtime</scope>-->
+<!--            </dependency>-->
+
```
Comment (AlbumenJ) [inline dubbo-dependencies-bom/pom.xml:265]: remove unused code

File: dubbo-remoting/dubbo-remoting-api/pom.xml [hunk 2]
```diff
@@ -42,7 +42,25 @@
...
+        </dependency>
+        <dependency>
+            <groupId>io.netty</groupId>
+            <artifactId>netty-transport</artifactId>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.netty</groupId>
+            <artifactId>netty-transport-native-epoll</artifactId>
+            <version>4.1.72.Final</version>
```
Comment (AlbumenJ) [inline dubbo-remoting/dubbo-remoting-api/pom.xml:61]: version should be managed in dubbo-dependenies-bom

File: dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/stub/StubInvocationUtilTest.java [hunk 3]
```diff
@@ -1,299 +0,0 @@
...
-            @Override
-            public void onCompleted() {
-                latch.countDown();
-            }
-        };
-        StubInvocationUtil.serverStreamCall(invoker, method, request, responseObserver);
-        Assertions.assertTrue(latch.await(1, TimeUnit.SECONDS));
-    }
-
-}
```
Comment (AlbumenJ) [inline dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/stub/StubInvocationUtilTest.java:299]: Why remove this class

File: dubbo-rpc/dubbo-rpc-triple/src/test/java/org/apache/dubbo/rpc/stub/StubInvocationUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+// * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// * See the License for the specific language governing permissions and
+// * limitations under the License.
+// */
+//
+//package org.apache.dubbo.rpc;
+//
+//import org.apache.dubbo.common.URL;
+//import org.apache.dubbo.common.stream.StreamObserver;
+//import org.apache.dubbo.rpc.model.ConsumerModel;
```
Comment (AlbumenJ) [inline dubbo-rpc/dubbo-rpc-triple/src/test/java/org/apache/dubbo/rpc/stub/StubInvocationUtilTest.java:22]: Why remove this class

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-remoting/dubbo-remoting-api/pom.xml [hunk 5]
```diff
@@ -42,7 +42,24 @@
...
+        </dependency>
+        <dependency>
+            <groupId>io.netty</groupId>
+            <artifactId>netty-transport</artifactId>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.netty</groupId>
+            <artifactId>netty-transport-native-epoll</artifactId>
+            <classifier>linux-x86_64</classifier>
```
Comment (AlbumenJ) [inline dubbo-remoting/dubbo-remoting-api/pom.xml:61]: should this filed change to os auto detected

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Comment by EarthChen** [comment]
`dubbo-all` is not modified for user compatibility

**Review by AlbumenJ — APPROVED** [review]
LGTM

**Pull request merged**